src/graphnet/data/datamodule.py

```diff
@@ -0,0 +1,338 @@
+from typing import Dict, Any, Optional, List, Tuple, Union
+import lightning as L
```

Comment:
Very cool that you're trying to get this to work. We are currently using the "older" syntax with pytorch_lightning and not lightning. I suspect that the run-time error that you're getting might a result of this.

Maybe you could try to do:
`import pytorch_lightning as pl`
and then also change where you've used `L.` Hope this works :)